nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/test/java/org/apache/nifi/processors/gcp/drive/ListGoogleDriveIT.java

```diff
@@ -0,0 +1,254 @@
+        expectedFileNames = new HashSet<>(Arrays.asList(
+                "main_sub1_file1"
+        ));
+
+        // WHEN
+        testRunner.setProperty(ListGoogleDrive.FOLDER_ID, main_sub1.getId());
+        testRunner.setProperty(ListGoogleDrive.RECURSIVE_SEARCH, "false");
+        testRunner.run();
+
+        // THEN
+        successFlowFiles = testRunner.getFlowFilesForRelationship(ListGoogleDrive.REL_SUCCESS);
+
+        actualFileNames = successFlowFiles.stream()
+                .map(flowFile -> flowFile.getAttribute("filename"))
+                .collect(Collectors.toSet());
+
+        assertEquals(expectedFileNames, actualFileNames);
+    }
+
+    @Test
+    void doNotListTooYoungFilesWhenMinAgeIsSet() throws Exception {
+        // GIVEN
+        testRunner.setProperty(ListGoogleDrive.MIN_AGE, "15 s");
+
+        createFile("main_file", mainFolderId);
+
+        // Make sure the file 'arrives' and could be listed
+        Thread.sleep(5000);
+
+        // WHEN
+        testRunner.run();
+
+        // THEN
+        List<MockFlowFile> successFlowFiles = testRunner.getFlowFilesForRelationship(ListGoogleDrive.REL_SUCCESS);
+
+        Set<String> actualFileNames = successFlowFiles.stream()
+                .map(flowFile -> flowFile.getAttribute("filename"))
+                .collect(Collectors.toSet());
+
+        assertEquals(Collections.emptySet(), actualFileNames);
+
+        // Next, wait for another 10+ seconds for MIN_AGE to expire then list again
+
+        // GIVEN
+        Thread.sleep(10000);
+
+        Set<String> expectedFileNames = new HashSet<>(Arrays.asList(
+                "main_file"
+        ));
+
+        // WHEN
+        testRunner.run();
+
+        // THEN
+        successFlowFiles = testRunner.getFlowFilesForRelationship(ListGoogleDrive.REL_SUCCESS);
+
+        actualFileNames = successFlowFiles.stream()
+                .map(flowFile -> flowFile.getAttribute("filename"))
+                .collect(Collectors.toSet());
+
+        assertEquals(expectedFileNames, actualFileNames);
+    }
+
+    private File createFolder(String folderName, String... parentFolderIds) throws IOException {
+        File fileMetaData = new File();
+        fileMetaData.setName(folderName);
+
+        if (parentFolderIds != null) {
+            fileMetaData.setParents(Arrays.asList(parentFolderIds));
+        }
+
+        fileMetaData.setMimeType("application/vnd.google-apps.folder");
+
+        Drive.Files.Create create = driveService.files()
+                .create(fileMetaData)
+                .setFields("id");
+
+        File file = create.execute();
+
+        return file;
+    }
+
+    private File createFile(String name, String... folderIds) throws IOException {
+        File fileMetadata = new File();
+        fileMetadata.setName(name);
+        fileMetadata.setParents(Arrays.asList(folderIds));
+
+        FileContent content = new FileContent("text/plain", new java.io.File("src/test/resources/ListGoogleDriveIT/test_file.txt"));
```

Comment:
Instead of checking in this file to version control, it would be better to create a temporary file that would be deleted after test completion.

Reply:
@exceptionfactory @tpalfy What do you think about `ByteArrayContent`?

Reply:
Good point @turcsanyip, using `ByteArrayContent` seems like a better solution than a local file path.